content/purgatorio/collect-and-prepare-data/data-collection.md

```diff
@@ -0,0 +1,209 @@
+
+[Back to the index](#index)
+
+### The Toolkit and sources
+
+Here is a list of things you could be doing to gather fresh data or existing data (and repurpose it for your needs):
+
+- Web Scrapping tools and services
+- Look for pre-existing datasets
+  - Kaggle Datasets
+  - Kaggle Kernels using custom datasets
+  - Other competition sites
+  - Other public domain sites
+  - Private domain sources (respecting the licenses)
+- Existing data in excel sheets or other forms within teams/organisations
+- Collecting log files of program or application executions from live systems
+- Running surveys via automatic services: Google Forms, SurveyMonkey, etc..,
+- Convert non-digitalised data using modern technology
+- Data generation services and tools
+
+Many of the above can be done in a "live" manner or through cron (scheduled) jobs and others through manual intervention.
+
+[Back to the index](#index)
+
+### Datasets 
+
+So the important question is what should the dataset look like? This will quite depend upon on your [Research or business questions](#Research-or-business-questions) to hand.
+
+Having a look at how others have created their datasets for the various domains or topics or research subjects, can also help in the process. See this [Datasets resource](https://github.com/neomatrix369/awesome-ai-ml-dl/blob/master/data/datasets.md) which is a mix of both clean and unclean datasets. This can also answer the more specific question, what file format should the dataset be saved in? CSV? JSON? or any other know file formats.
+
+They can throw light on types of data (features) to collect or not to collect, and in what form should they be in when collected such that it's useful for the next steps in the process.
+
+[Back to the index](#index)
+
+### Data Generation
+
+Data generation is also a good MVP step/process, and also could be incorporated as one of your steps during the first few iterations - if you discover or know from observation that the data to hand is not sufficient or correct or even not usable for the end-purpose (see [Research or business questions](#Research-or-business-questions)).
+
+It is often considered when a domain or field has limited data or previous experiments or exercises have resulted in a small dataset but the dataset may be some what representative of the domain or problem that is being solved - then Data Generation using existing could be take resort to. This of course has it's pros and cons - you soon get a lot of data but then is the data biased and in what way? Is it balanced or not, if not what to do - all these kinds of questions come to the surface.
```

Comment:
Split the first sentence for clarity if possible. 

Data Generation can be useful when a domain or field has limited data or previous experiments, or exercises have resulted in a small dataset, but the dataset may be somewhat representative of the problem that is being solved - in these cases Data Generation using existing data can help. This process is also called "synthetic data generation" if the data is generated from scratch, "data augmentation" if it's generated based on real pieces of data. 
See [Data Augmentation for Deep Learning](https://towardsdatascience.com/data-augmentation-for-deep-learning-4fe21d1a4eb9)
--->

It is often considered when a domain or field has limited data or previous experiments or exercises have resulted in a small dataset but the dataset may be somewhat representative of the domain or problem that is being solved - then Data Generation using existing could be take resort to.

Reply:
Reads much better than my version, I'll try to blend them together with small modifications!